Implemented ProductDTO for manifest import/export.

nikosmoum opened the pull request:
- ContentDTO was also added as a dependent of ProductDTO.
- Import and export endpoints do NOT use ProductDTO yet (this will be addressed in another task).
- Added comments with lists of fields we no longer export on ProductDTO, EntitlementDTO, ConsumerDTO and their nested DTOs.

Ceiu commented:
The code looks good, I just have reservations about some of the fields not being present -- specifically, the multiplier.

While it's true we just overwrite it today, the manifests themselves need to be forward and backward compatible (as much as that causes problems for us). I think this requires some closer inspection and discussion before we can remove them entirely.

nikosmoum commented:
@Ceiu I am unsure of what you mean here. Should I split this PR into two, where the first part is the ProductDTO and translators, and the second is their usage on the endpoints? Also, was there a discussion about the fields that were removed, and which of them should be re-added or not?

nikosmoum commented:
@Ceiu @vritant I previously wrongly assumed that the manifest DTOs should get the same treatment as the rules DTOs, in removing any fields that do not seem to be used. I've had a look over the previous manifest DTO PRs (CustomerDTO, EntitlementDTO and their nested DTOs), and gathered the complete list of fields that are no longer exported, which we should discuss about:
```
    - product.href
    - product.uuid
    - product.multiplier
    - product.productContent.content.uuid
    - product.productContent.content.metadataExpire

    - entitlement.href
    - entitlement.startDate
    - entitlement.endDate
    - entitlement.created
    - entitlement.updated

    - entitlement.certificates.created
    - entitlement.certificates.updated

    - entitlement.certificates.serial.revoked
    - entitlement.certificates.serial.serial

    - entitlement.pool.created
    - entitlement.pool.updated
    - entitlement.pool.type
    - entitlement.pool.owner
    - entitlement.pool.activeSubscription
    - entitlement.pool.createdByShare
    - entitlement.pool.hasSharedAncestor
    - entitlement.pool.sourceEntitlement
    - entitlement.pool.quantity
    - entitlement.pool.startDate
    - entitlement.pool.endDate
    - entitlement.pool.attributes
    - entitlement.pool.restrictedToUsername
    - entitlement.pool.consumed
    - entitlement.pool.exported
    - entitlement.pool.shared
    - entitlement.pool.calculatedAttributes
    - entitlement.pool.upstreamPoolId
    - entitlement.pool.upstreamEntitlementId
    - entitlement.pool.upstreamConsumerId
    - entitlement.pool.productAttributes
    - entitlement.pool.derivedProductAttributes
    - entitlement.pool.derivedProductName
    - entitlement.pool.stacked
    - entitlement.pool.stackId
    - entitlement.pool.developmentPool
    - entitlement.pool.sourceStackId
    - entitlement.pool.subscriptionSubKey
    - entitlement.pool.subscriptionId

    - entitlement.pool.branding.id (not to be confused with productId)
    - entitlement.pool.branding.created
    - entitlement.pool.branding.updated

    - consumer.owner.parentOwner
    - consumer.owner.key
    - consumer.owner.displayName
    - consumer.owner.contentPrefix
    - consumer.owner.lastRefreshed
    - consumer.owner.defaultServiceLevel
    - consumer.owner.upstreamConsumer
    - consumer.owner.logLevel
    - consumer.owner.contentAccessMode
    - consumer.owner.contentAccessModeList
    - consumer.owner.href
    - consumer.owner.created
    - consumer.owner.updated
    - consumer.owner.autobindDisabled
```

nikosmoum commented:
@candlepin-pull-request-bot retest this please

nikosmoum commented:
test this please

vritant commented:
@candlepin-pull-request-bot  test this please

nikosmoum commented:
@candlepin-pull-request-bot test this please

Ceiu commented:
Changes look good, but let's hold off on the merge until we discuss the "missing" fields for the manifest DTOs. It'll be easier to catch that early than to try to patch it back in later.

nikosmoum commented:
Re-added the following fields (but not product.href):
    - product.uuid
    - product.multiplier
    - product.productContent.content.uuid
    - product.productContent.content.metadataExpire

Ceiu commented:
ack
